ADL Updates for errors and new endpoints

begoldsm opened the pull request:
- Add DeleteAllSecrets (was a regression miss from hyak)
- Update Filesystem errors to reflect the types of errors thrown
- Fix json for InnerError -> innerError in job api (this is due to a fix finally happening on the server side)
- Add the ability to get partition information through the catalog APIs
- Add missing enum values
- Fix some documentation.

Commit:
Remove requirement on subscriptionId

These clients don't need a subscriptionId.

Commit:
Need to make contents required since this is a PUT.

Will add extensions that make it optional in C# and zero it out to an
empty stream.

Commit:
update documentation to reflect param required

Commit:
undo this temp change

Issue: https://github.com/Azure/autorest/issues/884 is tracking support
for this being an optional param properly and not resulting in a null
reference. So I will keep this spec as is.

Commit:
Change this name to be more descriptive.

Commit:
does not need to be modeled as a string.

Makes this more obvious to end users and easier to understand.

Commit:
fix all enums to be enums.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Add new API version for job

This API version changes the format for dates and timespan objects to be
ISO compliant.

Commit:
missed version change here as well.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
temp commit to remove the required flag

We need a way to have template parameters not be required when they are
in the host. I will work with Amar for this next week.

Commit:
Revert "temp commit to remove the required flag"

This reverts commit be8b7bf40e56c8316f423a871ccf030efe47a8a1.

Commit:
Add missing enum value

Commit:
Missed one.

Commit:
Add in new functionality

Switch over to MSLISTSTATUS which gives more flexibility for listing
files/folders.

Commit:
Work around for issue: 903

explicitly calling out our chunked transfer APIs until issue:
https://github.com/Azure/autorest/issues/903 is resolved.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
addressed comments from Amar.

Commit:
fix descriptions and add partition APIs

Commit:
sync from master

Commit:
Add missing method to delete all secrets in db

In hyak, we could do this by simply excluding the parameter. That is not
swagger compliant, so we need a separate entry.

Commit:
temporarily remove partition list for current ship.

this will be in the next snap.

Commit:
Add diagnostics to USqlJobProperties.

This was missing during the port from Hyak.

Commit:
Missing enum value.

Commit:
two more missing enums.

Commit:
Put partitions back in for next release.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Test out composites.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
add the log folder and logFilePattern to the sdk

This is for all jobs going forward.
Fix the readOnly properties to be bool instead of strings (makes tests
happy).

Commit:
Remove composite specs since they don't work

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Error json fixes

InnerError fixed to innerError
Adding the WebHDFS definition for ErrorResponses to the filesystem.

Commit:
bad syntax is bad.

Commit:
missed a boolean switch for readOnly,.

Commit:
definition -> definitions

Commit:
Change the name to be more descriptive

Commit:
Following feedback from the slack forum

This should help reduce the levels of nesting and also fixes some
feedback in the job swagger.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Add discriminators for webhdfs

Commit:
Further flatten and rename

Removing WebHdfsError and renaming "RemoteException"

Commit:
Rename to Adls

Commit:
nested exception discriminator

Commit:
update the required piece.

Commit:
try not indexing

Commit:
move it down to see if this fixes it.

Commit:
Try nesting again.

Commit:
Trying a work around.

Add a placeholder property at the root level which should get properly
picked up.

Commit:
Update to the work around

Slight change to the stub property.

Commit:
Work around didn't work, going back to what did.

Commit:
Removed the unneccessary nesting.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Add the accesscontrolexception

Commit:
Sync to latest master

amarzavery commented on arm-datalake-analytics/catalog/2015-10-01-preview/swagger/catalog.json:
```diff
@@ -976,6 +1002,153 @@
+              "$ref": "#/definitions/USqlTablePartitionList"
+            }
+          }
+        },
```
x-ms-odata ??

begoldsm replied:
Not sure what the question is here. We support x-ms-odata types for one type "USqlType", but that is not at this line. Do I need to include it everywhere?

amarzavery replied:
As per the [doc] (https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-odata) the **x-ms-odata** extension is applied on the operation and not on the type. This will indicate AutoRest that it needs to handle odata parameters in a special way. As this operation has odata parameters and the x-ms-odata extension was missing hence high highlighted it.
@markcowl, @hovsepm - is my understanding correct?

amarzavery commented on arm-datalake-analytics/catalog/2015-10-01-preview/swagger/catalog.json:
```diff
@@ -976,6 +1002,153 @@
+          }
+        ],
+        "responses": {
+          "200": {
```
"default" response?

begoldsm replied:
Is this a new requirement? by default, anything that is not a 200 response gets generated into a cloudException, which is currently what we want for everything except our filesystem swagger. What should I be putting in the default response if I want to continue to have the current behavior?

begoldsm replied:
@amarzavery for example, https://github.com/Azure/azure-rest-api-specs/blob/master/arm-resources/resources/2016-02-01/swagger/resources.json does not contain a single "default" either, since it uses the built in "everything that isn't explicitly specified is a CloudError" model.

amarzavery replied:
That should be fine

amarzavery commented on arm-datalake-analytics/catalog/2015-10-01-preview/swagger/catalog.json:
```diff
@@ -976,6 +1002,153 @@
+              "$ref": "#/definitions/USqlTablePartition"
+            }
+          }
+        }
```
general comment: have default response everyewhere

amarzavery replied:
As of today AutoRest will atleast deserialized code and message into the CloudError object. Any extra properties will not be deserialized since default does not have a schema. Although you will get the raw request and response as properties of the Exception so you can do deserialized it your way.

begoldsm replied:
@amarzavery that makes sense. Our errors actually adhere to the CloudError schema as it is right now, so it makes sense for us to use the default CloudException/CloudError setup. Below is the schema for all of our errors except Filesystem (which I updated). This should be properly deserialized by CloudError, right?

“error”: {
“code”: “string”,
“message”: “string”,
“target”: “optional string”,
“details”: [{
“code”: “string”,
“message”: “string”,
“target”: “optional string”
]},
                “innerError”: {
                                “context”: “string”,
                                “trace”: “sdf”
                                }
                }

amarzavery replied:
This will get deserialized [correctly in node.js as the mapper has it defined](https://github.com/Azure/autorest/blob/master/ClientRuntimes/NodeJS/ms-rest-azure/lib/cloudError.js#L60).
It looks like this will be [deserialized correctly in C#] as well(https://github.com/Azure/autorest/blob/master/ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime.Azure/CloudError.cs) (even when the default schema is not specified).
@markcowl @hovsepm - Is my understanding correct?

Commit:
remove extra next link stuff

I thought I had removed this previously. removing it again, since it
should no longer be used.

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
Remove currently unsupported API
